Mathlib/Algebra/Ring/Pi.lean

```diff
@@ -0,0 +1,184 @@
+! leanprover-community/mathlib commit ba2245edf0c8bb155f1569fd9b9492a9b384cde6
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+-/
+import Mathlib.Algebra.Group.Pi
+import Mathlib.Algebra.Hom.Ring
+
+/-!
+# Pi instances for ring
+
+This file defines instances for ring, semiring and related structures on Pi Types
+-/
+
+-- Porting notes: used to import `tactic.pi_instances`
+
+namespace Pi
+
+universe u v w
+
+variable {I : Type u}
+
+-- The indexing type
+variable {f : I → Type v}
+
+-- The family of types already equipped with instances
+variable (x y : ∀ i, f i) (i : I)
+
+-- Porting note: All these instances used `refine_struct` and `pi_instance_derive_field`
+
+instance distrib [∀ i, Distrib <| f i] : Distrib (∀ i : I, f i) :=
+  { add := (· + ·)
+    mul := (· * ·)
+    left_distrib := by intros; ext; exact mul_add _ _ _
+    right_distrib := by intros; ext; exact add_mul _ _ _}
+#align pi.distrib Pi.distrib
+
+instance nonUnitalNonAssocSemiring [∀ i, NonUnitalNonAssocSemiring <| f i] :
+    NonUnitalNonAssocSemiring (∀ i : I, f i) :=
+  { Pi.distrib, Pi.addCommMonoid, Pi.mulZeroClass with }
+#align pi.non_unital_non_assoc_semiring Pi.nonUnitalNonAssocSemiring
+
+instance nonUnitalSemiring [∀ i, NonUnitalSemiring <| f i] : NonUnitalSemiring (∀ i : I, f i) :=
+  { Pi.nonUnitalNonAssocSemiring, Pi.semigroupWithZero with }
+#align pi.non_unital_semiring Pi.nonUnitalSemiring
+
+instance nonAssocSemiring [∀ i, NonAssocSemiring <| f i] : NonAssocSemiring (∀ i : I, f i) :=
+  { Pi.nonUnitalNonAssocSemiring, Pi.mulZeroOneClass with }
+#align pi.non_assoc_semiring Pi.nonAssocSemiring
+
+instance semiring [∀ i, Semiring <| f i] : Semiring (∀ i : I, f i) :=
+  { Pi.nonUnitalSemiring, Pi.nonAssocSemiring, Pi.monoidWithZero with }
+#align pi.semiring Pi.semiring
+
+instance nonUnitalCommSemiring [∀ i, NonUnitalCommSemiring <| f i] :
+    NonUnitalCommSemiring (∀ i : I, f i) :=
+  { Pi.nonUnitalSemiring, Pi.commSemigroup with }
+#align pi.non_unital_comm_semiring Pi.nonUnitalCommSemiring
+
+instance commSemiring [∀ i, CommSemiring <| f i] : CommSemiring (∀ i : I, f i) :=
+  { Pi.semiring, Pi.commMonoid with }
+#align pi.comm_semiring Pi.commSemiring
+
+instance nonUnitalNonAssocRing [∀ i, NonUnitalNonAssocRing <| f i] :
+    NonUnitalNonAssocRing (∀ i : I, f i) :=
+  { Pi.addCommGroup, Pi.nonUnitalNonAssocSemiring with}
+#align pi.non_unital_non_assoc_ring Pi.nonUnitalNonAssocRing
+
+instance nonUnitalRing [∀ i, NonUnitalRing <| f i] : NonUnitalRing (∀ i : I, f i) :=
+  { Pi.nonUnitalNonAssocRing, Pi.nonUnitalSemiring with }
+#align pi.non_unital_ring Pi.nonUnitalRing
+
+instance nonAssocRing [∀ i, NonAssocRing <| f i] : NonAssocRing (∀ i : I, f i) :=
+  { Pi.nonUnitalNonAssocRing, Pi.nonAssocSemiring with}
+#align pi.non_assoc_ring Pi.nonAssocRing
+
+instance ring [∀ i, Ring <| f i] : Ring (∀ i : I, f i) :=
+  { Pi.semiring, Pi.addCommGroup, Pi.addGroupWithOne with}
```

Comment:
Untested,
```suggestion
  { Pi.semiring, Pi.addCommGroupWithOne with}
```

Reply:
`Algebra.Group.Pi` doesn't import `AddCommGroupWithOne`  and therefore is also not defining `Pi.addCommGroupWithOne`.